src/stores/deck.js

```diff
@@ -5,11 +5,8 @@ function generateDeck() {
   const suits = ['♠', '♥', '♦', '♣'];
   const ranks = ['A', '2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K'];
   const deck = [];
-  for (const suit of suits) {
-    for (const rank of ranks) {
-      deck.push({ suit, rank });
-    }
-  }
+  for (const suit of suits)
+    for (const rank of ranks) deck.push({ suit, rank, id: `${rank}${suit}` });
```

Comment:
We can’t use rank and suit for ID because custom decks will contain duplicate cards. Also I don’t think the deck store should know anything about the shop. It’s simply an interface for reading from and writing to an array of cards. We should be able to write to the deck from the shop. Instead of keeping track of the shop cards in here we could add a property to the card data structure so that it identifies each card that was acquired in the shop accordingly. Also prob need to generate a uuid to use for id here instead of rank and suit.